build/Utils.cmake

```diff
@@ -151,3 +152,48 @@ function(extract_sources sources_file)
     endif()
   endif()
 endfunction()
+
+# Determine the path to buck2, taking in order of preference: explicitly
+# provided buck2 argument, system buck2 (if correct version), download/cache
+# buck2.
+function(resolve_buck2)
+  if(EXECUTORCH_ROOT)
+    set(executorch_root ${EXECUTORCH_ROOT})
+  else()
+    set(executorch_root ${CMAKE_CURRENT_SOURCE_DIR})
+  endif()
+
+  set(resolve_buck2_command
+    ${PYTHON_EXECUTABLE} ${executorch_root}/build/resolve_buck.py
+    --cache_dir=${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
I'd rather avoid putting this at the root of the cmake-out directory. Could carve out a new spot like `--cache_dir=${CMAKE_CURRENT_BINARY_DIR}/buck2-bin`